chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -722,7 +726,15 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             );
         }
 
-        self.check_transactions(block)
+        let prev_block_height = self
```

Comment:
please confirm that it's ok to get previous block index in `check_block`. I mean I assume that the index of previous block always exist if we got here

Reply:
I believe so. You can see another height querying in check_block_header to check the reorg depth.